### PR TITLE
Remove alias from types.Signature to []byte

### PR DESCRIPTION
**Pull request by triazo**

### CODE REVIEW

File: compatibility/siag_1.0_test.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func verifyKeysSiag_1_0(uc types.UnlockConditions, folder string, keyname string
 			if err != nil {
 				return err
 			}
-			txn.TransactionSignatures[j].Signature = types.Signature(sig[:])
+			txn.TransactionSignatures[j].Signature = sig[:]
```
Comment (DavidVorick) [inline compatibility/siag_1.0_test.go]: should be sufficient to use `sig`, instead of `sig[:]`

File: modules/wallet/siag.go [hunk 2]
```diff
@@ -161,7 +161,7 @@ func (w *Wallet) SendSiagSiafunds(amount types.Currency, dest types.UnlockHash,
 			if err != nil {
 				return types.Transaction{}, err
 			}
-			txn.TransactionSignatures[sigIndex].Signature = types.Signature(encodedSig[:])
+			txn.TransactionSignatures[sigIndex].Signature = encodedSig[:]
```
Comment (DavidVorick) [inline modules/wallet/siag.go]: same

File: types/signatures_test.go [hunk 3]
```diff
@@ -249,13 +249,13 @@ func TestTransactionValidSignatures(t *testing.T) {
...
 	sig0[0]++
-	txn.TransactionSignatures[0].Signature = Signature(sig0[:])
+	txn.TransactionSignatures[0].Signature = sig0[:]
 	err = txn.validSignatures(10)
 	if err == nil {
 		t.Error("Corrupted a signature but the txn was still accepted as valid!")
 	}
 	sig0[0]--
-	txn.TransactionSignatures[0].Signature = Signature(sig0[:])
+	txn.TransactionSignatures[0].Signature = sig0[:]
```
Comment (DavidVorick) [inline types/signatures_test.go]: and same with all the other diffs :P

**Comment by lukechampine** [comment]
I ran some checks in the Go playground and it does indeed seem to be an alias problem, and specifically a []byte alias problem (string aliases are fine). Bummer.

**Comment by triazo** [comment]
Removed the slice copying.

**Comment by triazo** [comment]
Turns out that they were necessary for compilation, so added back in.

**Comment by DavidVorick** [comment]
oh

**Pull request merged**